File: androidMixFlutterDemo/my_android/app/src/main/java/com/jdt/myapplication/MainActivity.java
package com.jdt.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import io.flutter.embedding.android.FlutterActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button myButton = (Button) findViewById(R.id.myButton);
        myButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(
                        FlutterActivity.createDefaultIntent(MainActivity.this)
                );
            }
        });
    }
}

File: KotlinDemo/app/src/main/java/com/tgw/myapplication/MainActivity.kt
package com.tgw.myapplication

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModel
import com.tgw.myapplication.databinding.ObservableFieldActivity
import com.tgw.myapplication.databinding.ViewModelActivity


class MainActivity : AppCompatActivity() {

    @SuppressLint("JavascriptInterface")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val rollButton = findViewById<Button>(R.id.roll_button)
        rollButton.setOnClickListener{
            startActivity(Intent(this@MainActivity, DiceActivity::class.java))
        }

        val aboutMeButton = findViewById<Button>(R.id.aboutme_button)
        aboutMeButton.setOnClickListener{
            startActivity(Intent(this@MainActivity, AboutMeActivity::class.java))
        }

        val constraintButton = findViewById<Button>(R.id.constraint_button)
        constraintButton.setOnClickListener{
            startActivity(Intent(this@MainActivity, ConstraintLayoutActivity::class.java))
        }

        val observableButton = findViewById<Button>(R.id.observable_fields)
        observableButton.setOnClickListener {
            startActivity(Intent(this@MainActivity, ObservableFieldActivity::class.java))
        }

        val liveDataButton = findViewById<Button>(R.id.livedata_fields)
        liveDataButton.setOnClickListener {
            startActivity(Intent(this@MainActivity, ViewModelActivity::class.java))
        }
    }
}